src/libs/antares/study/area/links.cpp

```diff
@@ -406,20 +469,64 @@ bool AreaLinksInternalLoadFromProperty(AreaLink& link, const String& key, const
         link.filterYearByYear = stringIntoDatePrecision(value);
         return true;
     }
+    return false;
+}
+
+bool handleTSGenKey_internal(const std::string& key,
+                             const String& value,
+                             const std::string& prefix,
+                             Data::LinkTsGeneration& out)
+{
+    const auto checkPrefixed = [&prefix, &key](const std::string& s) {
+        auto key_lowercase(key);
+        boost::to_lower(key_lowercase);
+        return key_lowercase == prefix + "_" + s;
+    };
+
+    if (checkPrefixed("unitcount"))
+        return value.to<uint>(out.unitCount);
+
+    if (checkPrefixed("nominalcapacity"))
+        return value.to<double>(out.nominalCapacity);
+
+    if (checkPrefixed("law.planned"))
+        return value.to(out.plannedLaw);
+
+    if (checkPrefixed("law.forced"))
+        return value.to(out.forcedLaw);
+
+    if (checkPrefixed("volatility.planned"))
+        return value.to(out.plannedVolatility);
 
+    if (checkPrefixed("volatility.forced"))
+        return value.to(out.forcedVolatility);
+
+    return false;
+}
+
+bool handleTSGenKey(Data::AreaLink& link, const std::string& key, const String& value)
```

Comment:
Now arg **key** is an **std::string &**.
Do we need to define **key_** ?
If not, we should remove `const std::string key_(key);`

Reply:
[done](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2019/commits/07278105c09907bf73a2d219f38cfb034f09e708)